onboarding/main.go

```diff
@@ -57,90 +54,76 @@ func main() {
+	storageCluster, err := getStorageCluster(ctx, operatorNamespace, cl)
 	if err != nil {
-		klog.Errorf("failed to get storage cluster metadata: %v", err)
-		os.Exit(1)
-	}
-
-	privateSecret := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      onboardingValidationPrivateKeySecretName,
-			Namespace: operatorNamespace,
-			OwnerReferences: []metav1.OwnerReference{
-				{
-					UID:        storageClusterMetadata.UID,
-					APIVersion: storageClusterMetadata.APIVersion,
-					Kind:       storageClusterMetadata.Kind,
-					Name:       storageClusterMetadata.Name,
-				},
-			}},
-		StringData: map[string]string{
-			"key": privatePem,
-		},
-	}
-
-	_, err = clientset.CoreV1().
-		Secrets(operatorNamespace).
-		Create(ctx, privateSecret, metav1.CreateOptions{})
+		klog.Exitf("failed to get storage cluster: %v", err)
+	}
 
+	err = controllerutil.SetOwnerReference(storageCluster, privateSecret, cl.Scheme())
 	if err != nil {
-		klog.Errorf("failed to create private secret: %v", err)
-		os.Exit(1)
-	}
-
-	publicSecret := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      onboardingValidationPublicKeySecretName,
-			Namespace: operatorNamespace,
-			OwnerReferences: []metav1.OwnerReference{
-				{
-					UID:        storageClusterMetadata.UID,
-					APIVersion: storageClusterMetadata.APIVersion,
-					Kind:       storageClusterMetadata.Kind,
-					Name:       storageClusterMetadata.Name,
-				},
-			}},
-		StringData: map[string]string{
-			"key": publicPem,
-		},
-	}
-
-	_, err = clientset.CoreV1().
-		Secrets(operatorNamespace).
-		Create(ctx, publicSecret, metav1.CreateOptions{})
+		klog.Exitf("failed to set owner reference for privateSecret : %v", err)
+	}
+
+	privateSecret.StringData = map[string]string{
+		"key": privatePem,
+	}
+
+	err = cl.Create(ctx, privateSecret, &client.CreateOptions{})
 	if err != nil {
-		klog.Errorf("failed to create public secret: %v", err)
-		os.Exit(1)
+		klog.Exitf("failed to create private secret: %v", err)
+	}
+
+	err = controllerutil.SetOwnerReference(storageCluster, publicSecret, cl.Scheme())
+	if err != nil {
+		klog.Exitf("failed to set owner reference for public secret: %v", err)
+	}
+	publicSecret.StringData = map[string]string{
+		"key": publicPem,
+	}
+
+	err = cl.Create(ctx, publicSecret, &client.CreateOptions{})
+	if err != nil {
+		klog.Exitf("failed to create public secret: %v", err)
 	}
 
 }
 
-func newClient() (*kubernetes.Clientset, error) {
-	config := runtime.GetConfigOrDie()
-	clientset, err := kubernetes.NewForConfig(config)
+func newClient() (client.Client, error) {
+	klog.Info("Setting up k8s client")
+	scheme := runtime.NewScheme()
+	if err := v1.AddToScheme(scheme); err != nil {
+		klog.Errorf("Failed to add v1 to scheme: %v", err)
+	}
+	if err := corev1.AddToScheme(scheme); err != nil {
+		klog.Errorf("Failed to add corev1 to scheme: %v", err)
+	}
+	config, err := config.GetConfig()
 	if err != nil {
-		return nil, err
+		klog.Errorf("Failed to get config: %v", err)
```

Comment:
You should return here with an error. There is no point to log here, let the call site log the error